### DISCUSSION TITLE
Weekly Report 11.04-18.04

### DISCUSSION THREAD

**Discussion by Umbranecis** [SE-TINF22B6/Plapy#121] [Weekly reports]
This week, we used to complete what was left from last semester and start the new projects we want to work on in this semester.
For this, we finished our database setup to make sure of the following:

- Songs now get saved into the database and have a direct functional dependency to a playlist.
- When adding a song to a playlist, the playlist and song now gets updated in the database instantly, no need to call a save function any more
- The Playlist and songs now no longer get loaded from the database when starting the server, but only when called.
- The database structure now gets automatically created by typeORM framework so we doing need to manually assure the database structure is correct with our typescript files.

Further more, we implemented an extension for google Chrome to add songs directly fro YouTube to the bot.

Finally, we created an empty module that runs with react dependencies and will later include the website used to control our bot. This will be the main project for this year.

**Comment by Stev3eM** [comment]
Your project progress this week sounds impressive! You've tackled some key aspects, like finalizing last semester's tasks and laying the groundwork for the final part this semester. 

I especially like how you've optimized the database setup, ensuring songs and playlists are seamlessly connected and updated instantly. 

And the addition of a Google Chrome extension for YouTube song is a smart move for efficiency. 

It's great to see you already setting up the foundation for your main project this year with the React module. 

Keep up the good work!

<br>
<br>

<img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/ffa3f6d2-0d92-441e-9d78-51a62202b6e2" alt="carlos" align="right" width="100" />

**Your**   <img width="50" alt="image" src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/1a218c37-d3b0-4c88-9ae1-827b5a4eba42">   **Team**
